internal/client/client.go

```diff
@@ -180,15 +180,16 @@ func (c *Client) Shutdown(ctx context.Context) error {
 		ctx = context.Background()
 	}
 	c.EventDispatcher.Publish(lifecycle.NewLifecycleStateChanged(lifecycle.StateShuttingDown))
+	// execute registered shutdown handlers
+	for _, f := range c.shutdownHandlers {
+		f(ctx)
+	}
```

Comment:
Running shutdown handlers here changes the behaviour. Does this change has a reason?

Reply:
Resolved, [0042502](https://github.com/hazelcast/hazelcast-go-client/pull/772/commits/0042502d165af4bbfd57b99474f816dd147c4599).